tests/test_gnfa.py

```diff
@@ -387,9 +387,46 @@ def test_to_regex(self) -> None:
             # Test equality through DFA regex conversion
             self.assertEqual(dfa_1, dfa_2)
 
+    def test_gnfa_deterministic(self) -> None:
+        """
+        Check for deterministic conversion to GNFA.
+        From https://github.com/caleb531/automata/issues/231
+        """
+
+        gt_symbols = {"Connect": "0", "Send_msg": "1", "Ack": "2", "Close": "3"}
+        gt_dfa = DFA.from_nfa(
+            NFA(
+                states={"q0", "q1", "q2"},
+                input_symbols={
+                    gt_symbols["Connect"],
+                    gt_symbols["Send_msg"],
+                    gt_symbols["Close"],
+                    gt_symbols["Ack"],
+                },
+                transitions={
+                    "q0": {gt_symbols["Connect"]: {"q1"}},
+                    "q1": {gt_symbols["Close"]: {"q0"}, gt_symbols["Send_msg"]: {"q2"}},
+                    "q2": {gt_symbols["Ack"]: {"q1", "q0"}},
+                },
+                initial_state="q0",
+                final_states={"q0", "q1", "q2"},
+            )
+        )
+
+        # Repeat the test multiple times to account for possible non-determinism.
+        num_reps = 1_000
+        starting_gnfa = GNFA.from_dfa(gt_dfa)
+
+        for _ in range(num_reps):
+            regex = starting_gnfa.to_regex()
+            self.assertEqual(
+                "(012(012|12)*(03|3)|03)*(012(012|12)*(01?|1?)|(01?)?)", regex
+            )
+
     def test_show_diagram(self) -> None:
         """
         Should construct the diagram for a GNFA.
+
```

Comment:
@eliotwrobson Is there a reason this blank line was added to the docstring?

```suggestion
```

Reply:
Done by mistake I think. I'll remove. Also converting back to draft for now, since apparently didn't fix the issue completely.